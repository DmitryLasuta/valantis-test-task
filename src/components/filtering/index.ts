export * from './Filtering';
export * from './FilterField';
