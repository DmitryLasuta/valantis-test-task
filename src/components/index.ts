export * from './pagination';
export * from './ui/productCard';
export * from './Products';
export * from './filtering';
