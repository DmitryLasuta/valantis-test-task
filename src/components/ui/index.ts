export * from './grid';
export * from './pageLayout';
export * from './theFooter';
