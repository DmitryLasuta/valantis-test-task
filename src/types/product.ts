export interface Product {
  id: number;
  brand?: string;
  price?: number;
  product?: string;
}

export type ProductFieldNames = keyof Product;
export type ProductExcludingId = Omit<Product, 'id'>;
