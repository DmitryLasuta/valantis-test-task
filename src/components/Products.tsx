import { Grid } from './ui';
import { ProductCard } from './ui/productCard';
import { QUERY_KEYS } from '@/shared/constants';
import { storeAPI } from '@/shared/api/store';
import { useQuery } from '@tanstack/react-query';

const Products = ({ identifiers }: { identifiers: string[] }) => {
  const {
    data: products,
    isPending,
    isError,
  } = useQuery({
    queryKey: [QUERY_KEYS.PRODUCTS, identifiers],
    queryFn: () => storeAPI.getProducts(identifiers),
    staleTime: 1000 * 60 * 5,
  });

  if (!products || isPending) return <div>Loading...</div>;
  if (isError) return <div>Error</div>;

  return (
    <Grid>
      {products.map(product => (
        <ProductCard key={product.id} product={product} />
      ))}
    </Grid>
  );
};

export { Products };
