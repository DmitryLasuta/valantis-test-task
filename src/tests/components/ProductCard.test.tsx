import { describe, expect, test } from 'vitest';

import { Product } from '@/types';
import { ProductCard } from '@/components';
import { render } from '@testing-library/react';

describe('ProductCard', () => {
  test('should render correctly with initial props', () => {
    const testProduct: Product = {
      id: 1,
      price: 1000,
      brand: 'test brand',
      product: 'test product',
    };
    const { baseElement } = render(<ProductCard product={testProduct} />);

    expect(baseElement).toBeInTheDocument();
    expect(baseElement).matchSnapshot();
  });
});
