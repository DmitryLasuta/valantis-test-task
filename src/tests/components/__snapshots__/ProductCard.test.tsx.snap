// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductCard > should render correctly with initial props 1`] = `
<body>
  <div>
    <article
      class="_product-card_a15228"
    >
      <header>
        <h3
          class="_product-card__name_a15228"
        >
          test product
           | ID: 
          1
        </h3>
        <p
          class="_product-card__brand_a15228"
        >
          Brand: 
          test brand
        </p>
      </header>
      <p
        class="_product-card__price_a15228"
      >
        Price: $
        1000
      </p>
    </article>
  </div>
</body>
`;
